gtk/gtkfilechooserwidget.c

```diff
@@ -4056,6 +4098,8 @@ gtk_file_chooser_widget_map (GtkWidget *widget)
   settings_load (impl);
 
   add_cwd_to_sidebar_if_needed (impl);
+  show_places_button_if_needed (impl);
+  show_filters_scroll_if_needed (impl);
```

Comment:
this will only works if you add the extra widget before showing the filechooser widget.
I think you also need to call it in update_extra_and_filters() for example or better yet in extra_and_filters size-allocate signal